sample_configs/sample_config.yml

```diff
@@ -1,39 +1,27 @@
-liquid_hostname: liquidnode.liquid # defaults to liquidnode.liquid
-
 network:
-  hotspot: # hotspot is always on (Primary node WiFi/'admin' hotspot interface)
-    ssid: foo
-    password: bar
-    dhcp_server:
-      ip: 10.0.0.1 # defaults to 10.0.0.1
-      netmask: 255.255.255.0 # defaults to 255.255.255.0
-      dhcp_range: some_magic_dnsmasq_string? # defaults to 10.0.0.100-255
-  
-  lan: # for wired config of dhcp/dns server (Primary node)
-    interface: ethernet | none # no default
-    mode: dhcp_server | static # default to dhcp client
-    dhcp_server:
-      ip: 10.0.0.1 # defaults to 10.0.0.1
-      netmask: 255.255.255.0 # defaults to 255.255.255.0
-      dhcp_range: some_magic_dnsmasq_string? # defaults to 10.0.0.100-255
-    static:
-      ip: 192.168.66.66  # no default
-      netmask: 255.255.255.0 # no default
-      gateway: 192.168.66.1 # no default
-      dns_server: 192.168.66.1 # no default
-      
-  wan: # (Secondary node config or connection to internet)
-    interface: ethernet | wifi | none # no default
-    wifi:
+  domain: liquidnode.liquid # defaults to liquidnode.liquid
+
+  lan: # downstream network. clients connect and expect dhcp+dns from us.
+    ip: 10.0.0.1 # defaults to 10.0.0.1
+    netmask: 255.255.255.0 # defaults to 255.255.255.0
+    dhcp_range: 10.0.0.100-255 # defaults to 10.0.0.100-255
+
+    hotspot: # hotspot is always on (Primary node WiFi/'admin' hotspot interface)
       ssid: foo
       password: bar
-    mode: dhcp_client | static # default to dhcp client
-    static:
+
+    eth: false # if true, eth0 will function as dhcp+dns server; defaults to false
+
+  wan: # upstream network. we expect dhcp server (or static config) and optionally routing to the Internet.
+    static: # if missing, configure using dhcp
       ip: 192.168.66.66  # no default
       netmask: 255.255.255.0 # no default
       gateway: 192.168.66.1 # no default
       dns_server: 192.168.66.1 # no default
-  
+
+    wifi: # list of networks to connect as client
+      - ssid1: pass1
```

Comment:
First: why do you want a list here? This is the current networking configuration, let's keep it unambiguous.

I think a mechanism for remembering these passwords in the `liquid-core` UI & database (instead of here) would be better. Also, we want the user to be in complete control of what exact network they're connected on, so putting a list here without a `use this one` flag removes their choice.

Second: if you do it this way, the SSIDs will have to be unique, which is a unneeded limitation. We scan & try the ones that match anyway.

Reply:
> First: why do you want a list here?

I'm modelling the behaviour of laptops and smartphones. They keep a list of known networks and connect to them when they are in range. We can use a tool that implements this behaviour, like [wpa_cli](https://linux.die.net/man/8/wpa_cli).

> Second: if you do it this way, the SSIDs will have to be unique

No, because it's a yaml list of dictionaries, so you can have duplicate SSIDs.

Reply:
> yaml list of dictionaries
Gotcha. Okay.

Reply:
What's the diff between allowed key values in yaml and allowed SSIDs for wpa2?

We can work out trailing spaces with quotes, anything else that might be a problem?

Reply:
YAML keys: http://www.yaml.org/spec/1.2/spec.html#scalar//

> The content of a scalar node is an opaque datum that can be presented as a series of zero or more Unicode characters.

Reply:
Yaml can express any unicode string. For SSIDs, [they should be "octets"](https://en.wikipedia.org/wiki/Service_set_(802.11_network)#Service_set_identifier_.28SSID.29), and the encoding is not clear, but it's either latin1 or utf8, so no worries there. For passwords, [they should be "printable ascii characters"](https://en.wikipedia.org/wiki/Wi-Fi_Protected_Access#WPA_terminology), so again, no worries.